omegaconf/base.py

```diff
@@ -387,51 +417,81 @@ def _resolve_simple_interpolation(
 
     def _resolve_interpolation(
         self,
+        parent: Optional["Container"],
         key: Any,
         value: "Node",
         throw_on_missing: bool,
         throw_on_resolution_failure: bool,
     ) -> Any:
-        from .nodes import StringNode
+        value_kind, parse_tree = get_value_kind(value=value, return_parse_tree=True)
 
-        value_kind, match_list = get_value_kind(value=value, return_match_list=True)
-        if value_kind not in (ValueKind.INTERPOLATION, ValueKind.STR_INTERPOLATION):
+        if value_kind != ValueKind.INTERPOLATION:
             return value
 
-        if value_kind == ValueKind.INTERPOLATION:
-            # simple interpolation, inherit type
-            match = match_list[0]
+        assert parse_tree is not None
+        return self._resolve_complex_interpolation(
+            parent=parent,
+            value=value,
+            key=key,
+            parse_tree=parse_tree,
+            throw_on_missing=throw_on_missing,
+            throw_on_resolution_failure=throw_on_resolution_failure,
+        )
+
+    def resolve_parse_tree(
```

Comment:
It's computing the value of the expression represented by `parse_tree`.
This means essentially just calling the grammar visitor on the tree. This visitor requires three callbacks so that it can properly resolve all expressions:
- A callback to resolve node interpolations `${foo.bar}`
- A callback to evaluate resolvers `${resolver:x,yz}`
- A callback to evaluate quoted strings since they may also contain interpolations `"abc_${foo}"`

Reply:
roger.